lago/utils.py

```diff
@@ -561,3 +562,146 @@ def _add_subparser_to_cp(cp, section, actions, incl_unset):
         cp.set(section, var, str(action.default))
     if len(cp.items(section)) == 0:
         cp.remove_section(section)
+
+
+def with_validation(failure_msg):
```

Comment:
I think this decorator is problematic:
1. Use-case: it abstracts away the ability to handle the exceptions inside each function.
2. It relays on 'failure_on_error' to be defined on the decorated function, but it does not enforce setting it, so I would argue its less pythonic: explicit is better than implicit.
3. It relays that the function always returns the return code and specifically that it returns CommandStatus.

I think just updating 'run_command' with those parameters would be simpler: i.e.
```python
result = run_command(cmd=cmd, fail_on_error=True)
```